### PR TITLE
Add unittests for some simple system calls

### PR DESCRIPTION
**Pull request by LeeLin2602**
kill(2), wait4(2): add some testcase including different signals and waitpid wnohang.

### CODE REVIEW

File: etc/mtree/BSD.tests.dist [hunk 1]
```diff
@@ -933,6 +933,11 @@
         ..
         sys
         ..
+        syscalls
+            fork
+            ..
+            kill
```
Comment (emaste) [inline etc/mtree/BSD.tests.dist:939]: missing a `..` here?

File: tests/sys/syscalls/fork/fork_test.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <atf-c.h>
+#include <sys/wait.h>
+#include <sys/syscall.h>
+
+static int
+simple_test(int expected)
```
Comment (emaste) [inline tests/sys/syscalls/fork/fork_test.c:33]: arg probably better named `exit_status` - we're testing fork() with the child providing different return values (and, if we're going to test two values we should probably check 0 and some non-zero value)

Comment (LeeLin2602) [reply tests/sys/syscalls/fork/fork_test.c]: done

File: etc/mtree/BSD.tests.dist (hunk 1 above)
Comment (LeeLin2602) [reply etc/mtree/BSD.tests.dist:939]: done

**Comment by bsdimp** [comment]
These tests are a bit too basic. The don't test things that aren't already tested extensively elsewhere, and we don't test that various aspects of fork work (like inherited file descriptors, etc).

**Comment by LeeLin2602** [comment]
Hi, thank you for your comments.
But I would like to declare that this is the very first one step for our plan.
As you mentioned, this has been well tested in someplace else, however, this is the problem.
For a beginner, we cannot find the exact test file path quickly. And when we want to add more tests, we also have no idea where to append.
Therefore, this is our goal:
We are trying to test all the syscalls and would like to take it step by step.
So we finished the test for the most simple syscall. Hence, this is our very first step.
I hope you could understand it.
Thank you for your replying again.

**Pull request closed without merging**